### PR TITLE
refactor: DRY rsvpEvent

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

This flows slightly better, I think.

### CODE REVIEW

**Review by gikf — APPROVED** [review]
Yes it does :+1: 

One small nitpick, I'd think about pulling part below in the `else` of `if (oldEventUser)`
https://github.com/freeCodeCamp/chapter/blob/97077cb3faf93974ba1a4a8697dc40e21aee25a4/server/src/controllers/Events/resolver.ts#L341-L347

It's not going to stretch it vertically much, and after moving having two conditions in `if` could make it a bit easier to follow.

If you don't think this change is worth it, feel free to merge.

**Comment by ojeytonwilliams** [comment]
Cheers, that conditional was a bit ugly!

File: server/src/controllers/Events/resolver.ts [hunk 1]
```diff
@@ -325,28 +324,25 @@ export class EventResolver {
...
+        user: { connect: { id: ctx.user.id } },
+        event: { connect: { id: eventId } },
+        rsvp: { connect: { name: newRsvpName } },
+        event_role: { connect: { name: 'member' } },
+        subscribed: isSubscribedToChapter,
+      };
+      eventUser = await prisma.event_users.create({
+        data: eventUserData,
+        include: eventUserIncludes,
       });
```
Comment (gikf) [inline server/src/controllers/Events/resolver.ts:346]:
```suggestion
      });
      
      if (newRsvpName !== 'waitlist' && isSubscribedToChapter) {
        await createReminder({
          eventId,
          remindAt: sub(event.start_at, { days: 1 }),
          userId: ctx.user.id,
        });
      }
```
It needs to be moved after the `event_user` is created

**Comment by ojeytonwilliams** [comment]
Thanks @gikf it should all be okay now.  It bugs me that there's a (slightly) non-obvious ordering requirement, but I can live with it for now!

**Comment by gikf** [comment]
Well... can do!
```
      const eventUserData: Prisma.event_usersCreateInput = {
        user: { connect: { id: ctx.user.id } },
        event: { connect: { id: eventId } },
        rsvp: { connect: { name: newRsvpName } },
        event_role: { connect: { name: 'member' } },
        subscribed: isSubscribedToChapter,
        ...(newRsvpName !== 'waitlist' &&
          isSubscribedToChapter && {
            event_reminder: {
              create: {
                remind_at: sub(event.start_at, { days: 1 }),
              },
            },
          }),
      };
```

**Comment by ojeytonwilliams** [comment]
Yeaah, we could... I think the separation between reminders and events is useful, though.  Particularly given that we'll likely offload some of the reminding duties to Google.

**Review by gikf — APPROVED** [review]

**Pull request merged**